src/main/java/org/marketplace/controllers/AdvertisementImageManagementController.java

```diff
@@ -54,6 +61,7 @@ public Response<AdvertisementImage> requestUpdateAdvertisementImage(@RequestBody
      * Delete an advertisement image
      * @param id id of the advertisement image to be deleted
      */
+    @PreAuthorize("@resourceAccessAuthorizationService.authorizeUserAccess('ResourceType.AdvertisementImage', #id).equals(T(org.marketplace.enums.AccessStatus).ACCESS_GRANTED)")
```

Comment:
maybe i didn't understand this correctly, why we don't want to do something like?

```
@PreAuthorize("hasAuthority('SOME_AUTHORITY')")
```

Reply:
We discussed this issue during today's lecture.
As a followup you proposed some different kind of approach that links the user and his created resource for management access.

Could you explain it further with some ideas that we can read about?

Reply:
sure, it's called domain object security when we answer on 3 questions, who, where (method level), what (object level), ill try to come up with some example for you 👍

Reply:
i just created an example of a branch [feature/dima/acl](https://github.com/dawba/market-place/tree/feature/dima/acl), i am afraid it requires a bit more refactoring as it doesn't work with current securityconfig.
here is some general info about this approach https://github.com/eugenp/tutorials/tree/master/spring-security-modules/spring-security-acl